### ISSUE TITLE
Model#export incorrect example for PDF vector export

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#170]
[`Sketchup::Model#export`](https://ruby.sketchup.com/Sketchup/Model.html#export-instance_method) has incorrect example for PDF vector export

The second argument is `show_summary` in error.
Should be (**including proper indentation**!):
```ruby
# Export pdf file on a PC, showing a summary when complete.
options_hash = {
  :show_summary => true,
  :output_profile_lines => false,
  :map_fonts => false,
  :model_units => Length::Meter
}
status = model.export( 'c:/my_export.pdf', options_hash )
```

Not really a doc issue, but If the second argument is not a hash, the method just returns `false` and fails.
This is dumb. The code should then assume a hash of: 
```ruby
{ :show_summary => args[1] }
```
... as if the method was called like:
```ruby
status = model.export( 'c:/my_export.pdf', :show_summary => true )
# or
status = model.export( 'c:/my_export.pdf', show_summary: true )
```

**Comment by Eneroth3** [comment]
I struggle to parse this sentence "The second argument is show_summary in error.".

What is wrong with the current documentation?

**Comment by DanRathbun** [comment]
Seems as though the argument error (which prompted this report) has since been fixed.

But the indentation is still not ideal (for several examples.)

---

Docs aside, this report also reveals an error of a silent fail if a hash is not passed (as the second argument) for at least the PDF filetype (perhaps others.)

**Which may be why this issue is still open ?**

**Comment by thomthom** [comment]
> Which may be why this issue is still open ?

Probably slipped past us when we fixed it. I'll close this one.

> Docs aside, this report also reveals an error of a silent fail if a hash is not passed (as the second argument) for at least the PDF filetype (perhaps others.)

Without an option hash it would use default settings. Don't see that as a fail.. ?

**Comment by DanRathbun** [comment]
> Without an option hash it would use default settings. Don't see that as a fail.. ?

Except that, I had said ...

> Not really a doc issue, but If the second argument is not a hash, the **method just returns false and fails**.

Meaning it fails to export the file. Since this was a documentation issue, I'll retest and open new bug issue if this export failure stills exists in 2020 version.

**Comment by DanRathbun** [comment]
Okay, ... I just retested on SU2020, and the PDF file is written in all cases whether the 2nd arg is a hash or a boolean or something completely not in line with docs. Passing an arbitrary string is ignored for example (or might be interpreted as a `true` value.)

I also retested in SU2017 and (although there were issues with the PDF render) the file was still exported.

**I do not see the silent fail I reported 3 years ago.**

@Eneroth3 What I think I meant now regarding the second argument `:show_summary` is that it is disregarded by the method on Windows. Ie, **no summary is shown** despite the value being `true`. (There is a quick flash on a very small dialog that opens and closes so fast that there is no chance to read it.)

Secondly, any file at the target location is overwritten without warning the user ! This deviates from the norm for UX behavior.

```ruby
model = Sketchup.active_model
show_summary = true
status = model.export('c:/tmp/my_export.pdf', show_summary)
```
... fails to show summary.

```ruby
model = Sketchup.active_model
opts = { :show_summary => true }
status = model.export( 'c:/tmp/my_export.pdf', opts )
```
... fails to show summary.
